manager/scheduler/filter.go

```diff
@@ -217,3 +217,48 @@ func (f *ConstraintFilter) Explain(nodes int) string {
 	}
 	return fmt.Sprintf("scheduling constraints not satisfied on %d nodes", nodes)
 }
+
+// PlatformFilter selects only nodes that run the required platform.
+type PlatformFilter struct {
+	supportedPlatforms []*api.Platform
+}
+
+// SetTask returns true when the filter is enabled for a given task.
+func (f *PlatformFilter) SetTask(t *api.Task) bool {
+	c := t.Spec.GetContainer()
+	if c != nil {
+		if len(c.SupportedPlatforms) > 0 {
+			// copy the platform information
+			for _, p := range c.SupportedPlatforms {
+				f.supportedPlatforms = append(f.supportedPlatforms, p)
+			}
```

Comment:
I don't see anywhere that `f.supportedPlatforms` is cleared, so I think this slice will just grow every time a task is scheduled, and the filter won't work as expected.

Also note that you can do `append(f.supportedPlatforms, c.SupportedPlatforms...)` instead of using a loop.

But wouldn't it be even better to just do `f.supportedPlatforms = c.SupportedPlatforms`?